server/instance_cloud_oauth.go

```diff
@@ -71,20 +72,53 @@ func (p *Plugin) installCloudOAuthInstance(rawURL, clientID, clientSecret string
 		return "", nil, err
 	}
 
-	instance := &cloudOAuthInstance{
-		InstanceCommon:   newInstanceCommon(p, CloudOAuthInstanceType, types.ID(jiraURL)),
-		MattermostKey:    p.GetPluginKey(),
-		JiraClientID:     clientID,
-		JiraClientSecret: clientSecret,
-		JiraBaseURL:      rawURL,
-		CodeVerifier:     params.CodeVerifier,
-		CodeChallenge:    params.CodeChallenge,
+	newInstance := &cloudOAuthInstance{
+		InstanceCommon: newInstanceCommon(p, CloudOAuthInstanceType, types.ID(jiraURL)),
+		MattermostKey:  p.GetPluginKey(),
+		JiraBaseURL:    rawURL,
+		CodeVerifier:   params.CodeVerifier,
+		CodeChallenge:  params.CodeChallenge,
 	}
-	if err = p.InstallInstance(instance); err != nil {
-		return "", nil, err
+
+	existingInstance, err := p.instanceStore.LoadInstance(types.ID(jiraURL))
+	if err != nil && !errors.Is(err, kvstore.ErrNotFound) {
+		return "", nil, errors.Wrapf(err, "failed to load existing jira instance. ID: %s", jiraURL)
+	}
+
+	// Handle backwards compatibility with existing JWT instance
+	if existingInstance != nil {
+		if existingInstance.Common().Type == CloudOAuthInstanceType {
+			oauthInstance, ok := existingInstance.(*cloudOAuthInstance)
+			if !ok {
+				return "", nil, errors.Wrapf(err, "failed to assert existing cloud-oauth instance as cloudOAuthInstance. ID: %s", jiraURL)
+			}
+
+			newInstance.JWTInstance = oauthInstance.JWTInstance
+			if newInstance.JWTInstance != nil {
+				p.API.LogDebug("Installing cloud-oauth over existing cloud-oauth instance. Carrying over existing saved JWT instance.")
+			} else {
+				p.API.LogDebug("Installing cloud-oauth over existing cloud-oauth instance. There exists no previous JWT instance to carry over.")
+			}
+		}
+
+		if existingInstance.Common().Type == CloudInstanceType {
```

Comment:
Do we need to make it an `else if` condition to avoid an extra check?

Reply:
I used regular ifs so that there could be whitespace between the blocks to separate the two cases, though I think it reads better as you suggest. Please take a look at my suggested change below and add to the PR if GH allows you to do so through the UI